src/main/java/com/synopsys/integration/alert/provider/blackduck/descriptor/BlackDuckProviderDescriptorConfig.java

```diff
@@ -49,13 +51,13 @@ public BlackDuckProviderDescriptorConfig(final BlackDuckTypeConverter databaseCo
 
     @Override
     public UIComponent getUiComponent() {
-        final ConfigField blackDuckUrl = new ConfigField("blackduckUrl", "Url", FieldType.READ_ONLY, true, false, FieldGroup.DEFAULT);
-        final ConfigField blackDuckApiKey = new ConfigField("blackDuckApiKey", "API Token", FieldType.PASSWORD_INPUT, true, true, FieldGroup.DEFAULT);
-        final ConfigField blackDuckTimeout = new ConfigField("blackDuckTimeout", "Timeout", FieldType.NUMBER_INPUT, true, false, FieldGroup.DEFAULT);
-        final ConfigField blackDuckProxyHost = new ConfigField("blackDuckProxyHost", "Host Name", FieldType.READ_ONLY, false, false, FieldGroup.DEFAULT, PROXY_GROUP);
-        final ConfigField blackDuckProxyPort = new ConfigField("blackDuckProxyPort", "Port", FieldType.READ_ONLY, false, false, FieldGroup.DEFAULT, PROXY_GROUP);
-        final ConfigField blackDuckProxyUsername = new ConfigField("blackDuckProxyUsername", "Username", FieldType.READ_ONLY, false, false, FieldGroup.DEFAULT, PROXY_GROUP);
-        final ConfigField blackDuckProxyPassword = new ConfigField("blackDuckProxyPassword", "ProxyPassword", FieldType.READ_ONLY, false, true, FieldGroup.DEFAULT, PROXY_GROUP);
+        final ConfigField blackDuckUrl = new ReadOnlyConfigField("blackduckUrl", "Url", true, false);
+        final ConfigField blackDuckApiKey = new PasswordConfigField("blackDuckApiKey", "API Token", true);
+        final ConfigField blackDuckTimeout = new NumberConfigField("blackDuckTimeout", "Timeout", true, false);
+        final ConfigField blackDuckProxyHost = new ReadOnlyConfigField("blackDuckProxyHost", "Host Name", false, false, FieldGroup.DEFAULT, PROXY_GROUP);
```

Comment:
Why do some of the Field classes require you set the group and others do not? If the group is the default then why not expose the ConfigField constructor ` public ConfigField(final String key, final String label, final String type, final boolean required, final boolean sensitive) {` in the implementing class? I would like to see more consistency between the usages of the different Fields